### PR TITLE
Rust bindings (validate only)

### PR DESCRIPTION
**Pull request by axic**
Depends on #530. Depends on #533.

- [x] Change bumpversion
- [x] Add CI for `cargo build` and `cargo test`
- [x] Add CI for `cargo fmt`
- [ ] Change readme

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -512,6 +512,51 @@ jobs:
...
+    docker:
+      - image: rust:1-buster
+    steps:
+      - checkout
+      - run:
+          name: Update environment
+          command: |
+            echo 'deb http://deb.debian.org/debian buster-backports main' >> /etc/apt/sources.list
+            apt -qq update
+            apt -yq install libclang-dev clang cmake --no-install-recommends
```
Comment (axic) [inline circle.yml:525]: Even though I add buster-backports, this still installs cmake 3.13 from buster, while buster-backports has 3.16: https://packages.debian.org/buster-backports/cmake

Comment (axic) [reply circle.yml:526]: Fixed.

File: bindings/rust/build.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+use std::env;
+use std::path::PathBuf;
+
+fn main() {
+    // This is the root directory.
+    let src = "../../";
+
+    let dst = Config::new(src)
+        .define("FIZZY_TESTING", "OFF")
+        .define("FIZZY_FUZZING", "OFF")
```
Comment (chfast) [inline bindings/rust/build.rs:19]: The `FIZZY_FUZZING` depends on `FIZZY_TESTING` so not needed here.

File: bindings/rust/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// Fizzy: A fast WebAssembly interpreter
```
Comment (chfast) [inline bindings/rust/src/lib.rs:1]: Still surprises me that naming a file `lib.rs` is considered good practice.

File: bindings/rust/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    fn validate_wasm() {
+        assert_eq!(
+            validate(&[0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00]),
+            true
+        );
+        assert_eq!(
+            validate(&[0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x01]),
+            false
+        );
+        assert_eq!(validate(&[0x00]), false);
```
Comment (chfast) [inline bindings/rust/src/lib.rs:25]: Add test with empty input.

File: circle.yml (hunk 1 above)
Comment (chfast) [reply circle.yml:526]: Why are you installing cmake twice? Also: alpine 3.12 has cmake 3.17.

File: circle.yml [hunk 5]
```diff
@@ -512,6 +512,53 @@ jobs:
...
+            rustfmt --version
+            cargo fmt --all -- --check
+      - run:
+          name: Build
+          command: cargo build
+      - run:
+          name: Test
+          command: cargo test
+
+  bindings-rust-asan-combined:
```
Comment (chfast) [inline circle.yml:541]: What's "combined" about this?

File: circle.yml [hunk 6]
```diff
@@ -512,6 +512,53 @@ jobs:
...
+      - checkout
+      - run:
+          name: Update environment
+          command: |
+            echo 'deb http://deb.debian.org/debian buster-backports main' >> /etc/apt/sources.list
+            apt -qq update
+            apt -yq -t buster-backports install cmake --no-install-recommends
+            apt -yq install libclang-dev clang cmake --no-install-recommends
+            rustup component add rustfmt
+            rustup update
```
Comment (chfast) [inline circle.yml:528]: Why is update needed?

File: bindings/rust/src/lib.rs (hunk 3 above)
Comment (axic) [reply bindings/rust/src/lib.rs:1]: It is not "good practice", but required. The name is determined by the one in `Cargo.toml` and `lib.rs` is the default entry point.

File: circle.yml (hunk 5 above)
Comment (axic) [reply circle.yml:541]: The name came from evmc, not sure what it had, can remove the combined.

File: bindings/rust/src/lib.rs (hunk 4 above)
Comment (axic) [reply bindings/rust/src/lib.rs:25]: Sure, though this is a copy of the test in `capi_test.cpp`, which does not have that either.

File: circle.yml (hunk 6 above)
Comment (axic) [reply circle.yml:528]: To update any of the dependencies (also those needed by rustfmt). This is mostly a noop if the docker image is up to date (as it should be).

Comment (chfast) [reply circle.yml:528]: This seems wrong to be to having docker image being updated and also updating here. But I will not protest.

**Review by chfast — APPROVED** [review]

File: circle.yml (hunk 6 above)
Comment (axic) [reply circle.yml]: Removed.

**Review by chfast — APPROVED** [review]
